File: back/swift/app/core/repository/GameUseCases.swift
import Foundation

protocol GameUseCases {
    func initializeGame() -> Result<GameType, Error>
    func getGame() -> Result<GameType, Error>
    func makeMove(row: Int, col: Int, player: PlayerType) -> Result<GameType, Error>
    func getNextPlayer() -> Result<PlayerType, Error>
    func checkForWinner() -> Result<PlayerType, Error>
    func checkForDraw() -> Result<Bool, Error>
}

File: back/node/app/core/repository/GameUseCases.ts
import { Result } from "../common/result/Result";
import {GameType} from "../model/GameType";
import {PlayerType} from "../model/PlayerType";

interface GameUseCases {
    initializeGame(): Promise<Result<GameType>>;
    getGame(): Promise<Result<GameType>>;
    makeMove(row: number, col: number): Promise<Result<GameType>>;
    getNextPlayer(): Promise<Result<PlayerType>>;
    checkForWinner(): Promise<Result<PlayerType>>;
    checkForDraw(): Promise<Result<boolean>>;
}